projects/soft_999_shicheng_lu_fw_103/src/main.c

```diff
@@ -0,0 +1,54 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "interrupt_def.h"
+#include "log.h"
+#include "wait.h"
+
+// LOG_DEBUG the adc converted value of context pin when called (context is A6 pin)
+void log_adc_a6(const GpioAddress *address, void *context) {
+  GpioAddress *pin_a6_addr = context;
+  uint16_t pin_a6_data = 0;
+  adc_read_converted_pin(*pin_a6_addr, &pin_a6_data);
+  LOG_DEBUG("adc converted pin A6 data: %d\n", pin_a6_data);
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_init();
+  adc_init(ADC_MODE_SINGLE);

```

Comment:
You will need to call `adc_set_channel_pin` after you've initialized your gpio pin for the adc.
With the new functions in the ADC library, what happens is that you pass in a pin and a channel is handled for you so you don't have to do it manually.